core/os/dir_access.cpp

```diff
@@ -244,7 +244,7 @@ DirAccess *DirAccess::open(const String &p_path, Error *r_error) {
 
 	DirAccess *da = create_for_path(p_path);
 
-	ERR_FAIL_COND_V(!da, NULL);
+	ERR_FAIL_COND_V_MSG(!da, NULL, "Cannot open directory '" + p_path + "'.");
```

Comment:
`create_for_path` doesn't open a directory, it creates a `DirAccess` with the filesystem access relevant to the path (system, resource, user). So a correct error would be `"Cannot create DirAccess for path '" + p_path "'."`.